src/Helper/ArgumentResolverTrait.php

```diff
@@ -27,7 +27,12 @@ private function resolveArgument(\ReflectionParameter $param, ContainerInterface
 	{
 		$type = $param->getType();
 		if (!$type instanceof \ReflectionNamedType) {
-			throw new NotResolvedException('Can\'t auto-wire complex types');
+			throw new NotResolvedException(sprintf(
+				"Can't auto-wire complex types.\nArgument: \"%s\"\nType: \"%s\"\nClass: \"%s\"",
+				$param->getName(),
+				'complex', // todo get proper type.
```

Comment:
Should this not be `(string)$param->getType()?->__toString(),` ?

Reply:
`__toString()` is deprecated so I don't want to use it.

https://www.php.net/manual/en/reflectiontype.tostring.php

Reply:
Hmm according to [this](https://github.com/php/php-src/commit/530a8a38540e36a485f341d487529ade2b5d9c98) the `__toString()` method is no longer deprecated and you can use it for all 8.*.